### PR TITLE
Added pacman target and enabled 32-bit builds

### PR DESCRIPTION
**Pull request by aledomu**
AppImage was untested, so it was deleted. Also changed the portable target in Windows to zip since it's less prone to error. <https://github.com/angrykoala/gaucho/issues/225>

### Issues Related

------

Please, before doing the PR make sure that you are following the intructions described in [CONTRIBUTING.md](https://github.com/angrykoala/gaucho/blob/master/CONTRIBUTING.md) and you have completed the following checklist

**Checklist**
- [x] Your PR is done against the `dev` branch
- [x] Your origin branch is updated with the latest changes from `dev`
- [x] All the tests and jshint are passing (`npm run test`)
- [x] Changelog is updated with a brief description of your changes (if required)
- [ ] Your changes have tests

Thanks for contributing to Gaucho

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,22 +40,25 @@
...
+    "dist-win": "npm run clean && electron-builder -w --x64 --ia32",
+    "dist-mac": "npm run clean && electron-builder -m"
   },
   "build": {
     "appId": "angrykoala.gaucho",
     "linux": {
       "target": [
         "deb",
         "rpm",
-        "AppImage"
```
Comment (angrykoala) [inline package.json]: Why did you remove AppImage build?

File: package.json [hunk 2]
```diff
@@ -40,22 +40,25 @@
     "build-electron-all": "npm run clean && npm run build-electron -- --platform=win32 && npm run build-electron -- --platform=linux && npm run build-electron -- --platform=darwin",
     "clean": "rm -rf builds config.json tasks.json coverage dist",
     "lint": "eslint *.js app test",
-    "dist": "npm run clean && electron-builder -wlm"
+    "dist": "npm run clean && electron-builder -wlm --x64 --ia32",
+    "dist-linux": "npm run clean && electron-builder -l --x64 --ia32",
+    "dist-win": "npm run clean && electron-builder -w --x64 --ia32",
+    "dist-mac": "npm run clean && electron-builder -m"
```
Comment (angrykoala) [inline package.json]: --x64 and --ia32 are not needed here?

File: package.json [hunk 3]
```diff
@@ -40,22 +40,25 @@
...
-        "AppImage"
+        "pacman"
       ],
       "category": "Utility"
     },
     "win": {
       "target": [
         "nsis",
-        "portable"
+        "zip"
```
Comment (angrykoala) [inline package.json]: Why did you remove portable?

File: package.json (hunk 2 above)
Comment (aledomu) [reply package.json]: It's just to build for both x64 and x86 architectures. By default it will only build for the architecture you are running at compile-time (usually x64).

File: package.json (hunk 1 above)
Comment (aledomu) [reply package.json]: It needed some metadata that wasn't present and kept failing.

File: package.json (hunk 3 above)
Comment (aledomu) [reply package.json]: Packaging in zip deploys a portable version as well, it just doesn't make a first-run executable, so you launch it directly and you don't depend on any external tool in the packaging process. Although seeing it now, I think it's a good idea to switch the order between nsis and zip, so if nsis fails (happens sometimes in Linux), at least the zip target is built.

File: package.json (hunk 1 above)
Comment (angrykoala) [reply package.json]: Is that error part of the errors in https://github.com/angrykoala/gaucho/wiki/Common-Issues ?

I didn't have that error, I would like to maintain AppImage support unless this is not fixable

File: package.json (hunk 3 above)
Comment (angrykoala) [reply package.json]: After trying, I can see that the generated versions are:
* win-unpacked: standalone executable in the folder, this is generated always, for every system in the build
* Gaucho setup (nsis), installer for windows
* `zip`: same as unpacked but in a zip

The portable version is a single file that can be executed, while indeed, you can use the unpacked version (or the zip directly), I would like to maintain the "portable" version as a true standalone single-portable-file

File: package.json (hunk 2 above)
Comment (angrykoala) [reply package.json]: Yeah, I know, my question is why not using it in `dist-mac`?

Comment (aledomu) [reply package.json]: macOS just works on 64 bits.

File: package.json (hunk 1 above)
Comment (aledomu) [reply package.json]: Nope, none of those. I could get it back and let you test it (if it worked for you previously, it should still work).

File: package.json (hunk 3 above)
Comment (aledomu) [reply package.json]: Ok, will change.

File: package.json (hunk 1 above)
Comment (angrykoala) [reply package.json]: Could you show me the error? (I already tested your version with AppImage and it seemed to work, I'll try again)

**Comment by angrykoala** [comment]
Please @aledomu , check the issue #238 for discussion about 32 bits

File: package.json (hunk 1 above)
Comment (aledomu) [reply package.json]: It shows me this:
`./dist/__appimage-x64 should be packaged as /home/AJ/hacktoberfest/gaucho/dist/gaucho-0.4.0-x86_64.AppImage`

File: package.json [hunk 4]
```diff
@@ -35,24 +35,27 @@
     "test": "istanbul cover _mocha -- --exit && npm run lint",
     "start": "electron .",
     "start-dev": "cross-env NODE_ENV=dev npm start",
-    "clean": "rm -rf builds config.json tasks.json coverage dist",
+    "clean": "rm -rf config.json tasks.json coverage dist",
     "lint": "eslint *.js app test",
-    "dist": "npm run clean && electron-builder -wlm"
+    "dist": "electron-builder",
+    "dist-all": "npm run clean && electron-builder --x64 --ia32"
```
Comment (angrykoala) [inline package.json]: A few concerns:
* What does `electron-builder` (without arguments) do?
* Why not adding npm run clean to dist?
* Don't we need the -wlm option in `dist-all`?

Comment (aledomu) [reply package.json]: 1. That `dist` script is there to build for all platform specified in `package.json` in the architecture currently running the script.
2. The `dist` script is thought to be more "manual" so to speak, it is meant to be used passing arguments so we don't have to make a script for each platform and/or target. Adding `npm run clean` would remove previous builds even if you are building to a platform not already built.
3. Not at all, it would package for all platforms and targets configured in `package.json`. Also this way you can specify to build to just one platform and/or target in both currently supported architectures, although it's discouraged because this script does remove all previous builds.

If you don't agree with the rationale, feel free to ask more questions or ask for specific fixes.

Comment (angrykoala) [reply package.json]: 1. Ok, I don't understand why, by default, `dist` only generates current OS build, while `dist-all` generates all by default without other arguments

2. I agree with passing arguments, but I'm not sure how those arguments are supposed to be passed, could you try and document this in the readme?

Comment (aledomu) [reply package.json]: I have tested it and I must have done something wrong, so dist-all has the same issue, my bad. So we do need `-wlm` in `dist-all`.
Passing arguments works this way, for example, `npm run dist -- --ia32 -w` launches the script and passes the arguments after `--`, so it builds the Windows targets in 32-bit, regardless of the OS you are running. I will document it.
